### ISSUE TITLE
Suggestion: Access to globals in macros (3.0) -> maybe with _context. appendix?

### ISSUE THREAD

**Issue by jochen777** [PebbleTemplates/pebble#242]
The fact, that macros don't have access to global vars is good in the first place in my opinion. It creates a scope and provides "information hiding". (And it is like in twig by the way)

But maybe sometimes this is not easy for developers and leads to long parameter lists.

What about prefixing the access to the global scope with - for example "GLOBAL."? So you can check quickly, that you access global vars.

 Or maybe better: You can pass the global scope with a special parameter named "_context". (like in twig)

Example

```html
{% macro input(type="text", name, value, _context) %}
    <input type="{{ type }}" name="{{ name }}" value="{{ value }}" />
    <input type="submit" value="{{ _context.submitName }}">
{% endmacro %}
```

**Comment by mbosecke** [comment]
I'm hoping to hear some more opinions on this matter; my own opinion is conflicted.

I also like that macros used to provide "information hiding" but in my experience this caused more harm than good. It wasn't flexible enough for my use cases. Perhaps using a special variable name is the appropriate middle ground but I've never been a fan of special variables; they always feel like a work-around to a problem but maybe that's just me.

Anyways, I haven't committed to any way in particular so I'm hoping someone else can share their experience/thoughts.

**Comment by ebussieres** [comment]
@jochen777 Can you check if #231 resolves this issue. I think it's a duplicate. Thanks

**Comment by jochen777** [comment]
Well, my ticket was a response to the plan, to allow access to global scope in macros. I'm still against this global scope in macros. So, no this does not resolves the issue. BUT if everybody else love the global scope, than go for it :)

**Comment by ebussieres** [comment]
Resolved via #376
